### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by rpkilby**

### CODE REVIEW

File: docs/ref/filterset.txt [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+.. _order-by:
+
+Ordering using ``order_by``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can allow the user to control ordering by providing the
+``order_by`` argument in the Filter's Meta class.  ``order_by`` can be either a
+``list`` or ``tuple`` of field names, in which case those are the options, or
+it can be a ``bool`` which, if True, indicates that all fields that
+the user can filter on can also be sorted on. An example or ordering using a list::
```
Comment (carltongibson) [inline docs/ref/filterset.txt]: ... example **of** ordering...

File: docs/ref/filters.txt [hunk 2]
```diff
@@ -3,6 +3,118 @@ Filter Reference
...
+argument.
+
+``widget``
+~~~~~~~~~~
+
+The django.form Widget class which will represent the ``Filter``.  In addition
+to the widgets that are included with Django that you can use there are
+additional ones that django-filter provides which may be useful:
+
+    * ``django_filters.widgets.LinkWidget`` -- this displays the options in a
```
Comment (carltongibson) [inline docs/ref/filters.txt]: We have more than this now. Perhaps just a quick definition list.

Should link to widget reference.

File: docs/ref/filters.txt [hunk 3]
```diff
@@ -3,6 +3,118 @@ Filter Reference
...
+      manner similar to the way the Django Admin does, as a series of links.
+      The link for the selected option will have ``class="selected"``.
+
+``action``
```
Comment (carltongibson) [inline docs/ref/filters.txt:44]: Only applies to MethodFilter. Should be listed there instead.

File: docs/ref/filters.txt [hunk 4]
```diff
@@ -3,6 +3,118 @@ Filter Reference
...
+``lookup_expr``
+~~~~~~~~~~~~~~~
+
+The lookup expression that should be performed using `Django's ORM`_.
+
+.. _`Django's ORM`: https://docs.djangoproject.com/en/dev/ref/models/querysets/#field-lookups
+
+A ``list`` or ``tuple`` of lookup types is also accepted, allowing the user to
+select the lookup from a dropdown. The list of lookup types are filtered against
+``filters.LOOKUP_TYPES``. If no lookups are passed, then a list of all lookup
```
Comment (carltongibson) [inline docs/ref/filters.txt]: If `lookup_expr=None` is passed...

File: docs/ref/filters.txt (hunk 3 above)
Comment (carltongibson) [inline docs/ref/filters.txt:44]: Looking at `Filter.__init__`, the `action` kwargs is there. I'm not sure it should be...

**Comment by carltongibson** [comment]
OK. A few inline comments but this is awesome! I think it makes for a lot clearer reading. Great work. Let's have it.

File: docs/ref/filters.txt (hunk 3 above)
Comment (rpkilby) [inline docs/ref/filters.txt:44]: I agree, but think it should be handled in a separate PR as it's backwards incompatible.

Comment (rpkilby) [inline docs/ref/filters.txt:44]: Although, it could actually be useful. A `NumberFilter` would perform the value validation while still passing off the actual filtering to `action`. Currently, `MethodFilter`s do not perform any field validation (would need to either subclass with a custom `field_class` or perform validation in the action itself.

I'd have to look into it further, but it might make sense to deprecate `MethodFilter` and merge the functionality back into the base filter class as just specifying an `action`. Take a look at the drf-filters implementation [here](https://github.com/philipn/django-rest-framework-filters/blob/master/rest_framework_filters/filters.py#L84-L125). It might need a little work, but the `resolve_action` hook could be called after the name is set [here](https://github.com/carltongibson/django-filter/blob/develop/django_filters/filterset.py#L43-L44).

Either way - probably better to discuss this in another issue.

File: docs/ref/filters.txt (hunk 2 above)
Comment (rpkilby) [inline docs/ref/filters.txt]: Added `BooleanWidget` and `CSVWidget`, as these two seem independently useful. The others only seem to have internal use.

**Comment by rpkilby** [comment]
Going ahead and merging this since there aren't any contentious issues.
